src/base/server_type.cpp

```diff
@@ -83,8 +83,18 @@ void finalize_server_type ()
 
 int init_server_type (const char *)
 {
-  g_server_type = SERVER_TYPE_TRANSACTION;
-  return NO_ERROR;
+  int err_code = NO_ERROR;
+
+  g_server_type = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
```

Comment:
SA_MODE still won't act as a page server. Better to leave it a transaction server, regardless of the system parameter. Later on, when transaction server will lose local storage, maybe there will be different server types - transaction and SA. To be determined.
```suggestion
  g_server_type = SERVER_TYPE_TRANSACTION;
```

Reply:
Done